cmd/build.go

```diff
@@ -437,43 +437,28 @@ func runKustomizeBuild(path string) (rm resmap.ResMap, err error) {
 	return rm, nil
 }
 
-func generateApplicationNodePoolOptions(nodePoolData []*yaml2.Node) string {
-	applicationNodePoolOptions := []string{"|\n"}
-	var optionChunk []string
-	var prefix string
-	var optionChunkAppend string
-	addSingleQuotes := false
-	for _, poolNode := range nodePoolData {
-		//Find the sequence tag, which refers to options
-		if poolNode.Tag == "!!seq" {
-			optionsNodes := poolNode.Content
-			for _, optionNode := range optionsNodes {
-				for idx, optionDatum := range optionNode.Content {
-					if idx%2 == 1 {
-						continue
-					}
-					prefix = "  " //spaces instead of tabs
-					if strings.Contains(optionDatum.Value, "name") {
-						prefix = "- "
-					}
-					if optionNode.Content[idx+1].Tag == "!!str" {
-						addSingleQuotes = true
-					}
-					if addSingleQuotes {
-						optionChunkAppend = "    " + prefix + optionDatum.Value + ": '" + optionNode.Content[idx+1].Value + "'\n"
-					} else {
-						optionChunkAppend = "    " + prefix + optionDatum.Value + ": " + optionNode.Content[idx+1].Value + "\n"
-					}
-					optionChunk = append(optionChunk, optionChunkAppend)
-					addSingleQuotes = false
-				}
-				optionChunk = append(optionChunk, "")
-			}
-			break
-		}
+func generateApplicationNodePoolOptions(nodePoolData *yaml2.Node) string {
+	nodePool := struct {
+		Options []map[string]interface{}
+	}{}
+	err := nodePoolData.Decode(&nodePool)
```

Comment:
Minor nitpick, but why not do 
``` if err := nodePoolData.Decode(&nodePool); err != nil {
		log.Println(err)
		return ""
}

Reply:
Good catch, will fix shortly.

Reply:
Only thing thing I would say is - we might want to add unit tests for things like this. There looks like a lot of logic in the old one, but I think you've got it covered with your decoding into a struct.

Reply:
Agreed

Reply:
Added test